.github/workflows/python-linters.yml

```diff
@@ -38,16 +69,33 @@ jobs:
       run: |
         git config --global user.email "pypa-dev@googlegroups.com"
         git config --global user.name "pip"
-    - name: Update setuptools and tox dependencies
-      run: |
+    - name: Update setuptools
+      run: >-
         python -m pip install --upgrade setuptools
+    - name: Install tox
+      run: >-
         python -m pip install --upgrade tox tox-venv
+    - name: Log the list of packages
+      run: >-
         python -m pip freeze --all
     - name: 'Initialize tox envs: ${{ matrix.env.TOXENV }}'
       run: >-
-        python -m tox --notest --skip-missing-interpreters false
+        python -m
+        tox
+        --parallel auto
+        --notest
+        --skip-missing-interpreters false
       env: ${{ matrix.env }}
+    - name: Pre-fetch pre-commit hooks
+      # This is to separate test step from deps install
+      if: matrix.env.TOXENV == 'lint'
+      run: >-
+        .tox/lint/${{ runner.os == 'Windows' && 'Scripts' || 'bin' }}/python -m
+        pre_commit
+        install-hooks
```

Comment:
Is this needed? `pre-commit` would do this on it's own on first run.

If it's not necessary, I'd rather that we drop it.

Reply:
It's good to separate install/init step from the actual test run because then it's clear what fails. It's the same as `tox --notest` + `tox` in a separate step.